website/src/content/docs/formatter/index.mdx

```diff
@@ -93,7 +93,138 @@ const expr =
+
+Input
+
+```js title="example.js"
+a = {
+  [x = 0]: 1,
+}
+
+class C {
+  [x = 0] = 1
+}
+```
+
+Diff
+
+```js title="example.js" del={2} ins={3}
+a = {
+  [(x = 0)]: 1,
+  [x = 0]: 1,
+};
+
+class C {
+  [x = 0] = 1;
+}
+```
+
+[Playground link](https://biomejs.dev/playground?enabledLinting=false&code=YQAgAD0AIAB7AAoAIAAgAFsAeAAgAD0AIAAwAF0AOgAgADEALAAKAH0ACgAKAGMAbABhAHMAcwAgAEMAIAB7AAoAIAAgACAAIABbAHgAIAA9ACAAMABdACAAPQAgADEACgB9AAoA)
+
+To be consistent, we decide to diverge and omit the parentheses.
+Alternatively, we could enclose any assignment in a computed key of an object or of a class.
+
+
+### Prettier adds a trailing comma to type parameters of arrow functions even when it is not required.
+
+In some specific cases, a type parameter list of an arrow function requires a trailing comma to distinguish it from a JSX element.
+When a default type is provided, this trailing comma is not required.
+Here, we diverge from Prettier, because we think it better respects the original intent of Prettier, which was to add a trailing comma only when required.
+
+Input
+
+```tsx title="example.tsx"
+<T = unknown>() => {};
+```
+
+Diff
+
+```tsx title="example.tsx" del={1} ins={2}
+<T = unknown,>() => {};
+<T = unknown>() => {};
+```
+
+
+### Prettier formats invalid syntaxes
+
+As mentioned in a previous divergence, Prettier's Babel-based parsing for JS/TS is very loose and allows multiple errors to be ignored.
+Biome's parser intentionally has stricter rules than Prettier and correctly rejects an assignment to an optional chain.
+Biome represents this invalid assignment using a bogus node and don't try to format the assignment.
+
+### Prettier formats invalid syntaxes
+
+Prettier's Babel-based parsing for JavaScript and TypeScript is very loose and [allows multiple errors](https://github.com/prettier/prettier/blob/e4a74c05f4502dd4ec70495c3130ff08ab088e05/src/language-js/parse/babel.js#L177-L218) to be ignored.
+Biome's parser is intentionally stricter than the Prettier parser.
+It correctly identifies the following syntax errors:
+
+- A function cannot have duplicate modifiers
+- invalid order of properties' modifiers
+- Function declarations are not allowed to have bodies
+- non-abstract classes cannot have abstract properties
+- An optional chain cannot be assigned
+- The `const` modifier cannot be set on a type parameter of an interface
+- top-level return
+- etc.
+
+In Prettier, these errors aren't considered parse errors, and the AST is still built "correctly" with the appropriate nodes.
+When formatting, Prettier then just treats these nodes as normal and formats them accordingly.
+
+In Biome, the parsing errors result in `Bogus` nodes, which may contain any number of valid nodes, invalid nodes, and/or raw tokens.
+When formatting, Biome treats bogus nodes as effectively plain text, printing them out verbatim into the resulting code without any formatting, since attempting to format could be incorrect and cause semantic changes.
```

Comment:
```suggestion
Prettier's Babel-based parsing for JavaScript and TypeScript is very loose and [allows multiple errors](https://github.com/prettier/prettier/blob/e4a74c05f4502dd4ec70495c3130ff08ab088e05/src/language-js/parse/babel.js#L177-L218) to be ignored.
Biome's parser is intentionally stricter than the Prettier parser.
It correctly identifies the following syntax errors:

- A function cannot have duplicate modifiers
- invalid order of properties' modifiers
- Function declarations are not allowed to have bodies
- non-abstract classes cannot have abstract properties
- An optional chain cannot be assigned
- The `const` modifier cannot be set on a type parameter of an interface
- top-level return
- etc.

In Prettier, these errors aren't considered parse errors, and the AST is still built "correctly" with the appropriate nodes.
When formatting, Prettier treats these nodes as normal and formats them accordingly.

In Biome, the parsing errors result in `Bogus` nodes, which may contain any number of valid nodes, invalid nodes, and/or raw characters.
When formatting, Biome treats bogus nodes as effectively plain text, printing them out verbatim into the resulting code without any formatting since attempting to format them could be incorrect and cause semantic changes.
```